docs/synthetics/puppeteer-scripts/single-page-app.js

```diff
@@ -1,12 +1,27 @@
 async function testPage(page) {
-    await page.goto('https://www.instagram.com/');
-    // wait for sign up link element to appear, since it is loaded asynchronously
-    await page.waitForSelector('a[href="/accounts/emailsignup/"]');
-    await page.click('a[href="/accounts/emailsignup/"]');
-    // again wait for login link to appear
-    await page.waitForSelector('a[href="/accounts/login/?source=auth_switcher"]');
-    await page.click('a[href="/accounts/login/?source=auth_switcher"]');
-    await page.waitForSelector('a[href="/accounts/password/reset/"]');
+    // Set the screen resolution, as things render differently on different screen widths
+    await page.setViewport({
+        width: 1200,
+        height: 800,
+    })
+
+    // Go to the URL which switches into the demo account and wait for the redirect to finish
+    await page.goto('https://apps.sematext.com/demo');
+    await page.waitForTimeout(3000);
+
+    const SYNTHETICS_APP_SELECTOR = 'a[href="/ui/synthetics"]';
+    await page.waitForSelector(SYNTHETICS_APP_SELECTOR);
+    await page.click(SYNTHETICS_APP_SELECTOR);
+
+    // Wait for the locations map to show up, then wait a bit more for it to load fully for the screenshot
+    await page.waitForSelector('[id="locationsMap"]');
+    await page.waitForTimeout(2500); // We only need this to load everything for the screenshot
```

Comment:
The timeout here is to wait for the visual parts to load in too, and it's also optional, hence the comment. I haven't tried using it since the current description for the example implies that it shouldn't be used after navigating to the starting URL. Maybe it changed and behaves differently now?
> Any navigation across the app does not trigger a new page load/navigation event. The script has to rely on the visibility/availability of elements for the navigation.

Reply:
SPA also register navigations. Our own app changes URL. It is just that the navigation event is hijacked by the router. Never the less all these navigations are part of browser  history as well so these very much should count as navigations. Try running with waitNavigation to see what happens.

Reply:
It's not really working in this case - runs always time out even when `waitForNavigation` is used right after the click which navigates to a different page

Reply:
Ok then merge away 🚢